### PR TITLE
Exercises for Ch.Syntax & Ch.Elaboration

### PR DESCRIPTION
**Pull request by lakesare**
[Based on #84]

### In this PR

- Created exercises & solutions for **Ch.Syntax**
- Created exercises & solutions for **Ch.Elaboration**
- Fixed missing code blocks in **Ch.MetaM** exercises
- in `/solutions` we used to have abridged descriptions of the exercise text - removed that, let's just have the exercise number

### Links to the latest versions of updated files

- **Ch.Syntax** [Exercises](https://github.com/lakesare/lean4-metaprogramming-book/blob/lakesare_exercises_2/md/main/syntax.md#exercises), [Solutions](https://github.com/lakesare/lean4-metaprogramming-book/blob/lakesare_exercises_2/md/solutions/syntax.md)

- **Ch.Elaboration** [Exercises](https://github.com/lakesare/lean4-metaprogramming-book/blob/lakesare_exercises_2/md/main/elaboration.md#exercises), [Solutions](https://github.com/lakesare/lean4-metaprogramming-book/blob/lakesare_exercises_2/md/solutions/elaboration.md)

### CODE REVIEW

File: lean/main/syntax.lean [hunk 1]
```diff
@@ -551,4 +553,71 @@ it in the way we would expect it.
...
+
+Use the following code template:
+
+```
+syntax (name := colors) ...
+-- our "elaboration function" that infuses syntax with semantics
+@[command_elab colors] def elabColors : CommandElab := λ stx => Lean.logInfo "success!"
+```
+
+4. Mathlib has a `#help option` command that displays all options available in the current environment, and their descriptions. `#help option pp.r` will display all options starting with a "pp.r" substring.
```
Comment (hargoniX) [inline lean/main/syntax.lean:590]: I find the task to be a little ambiguous here, people could think you want them to only  allow identifiers of options which is not possible without an elaborator, maybe make this a little more clear?

File: lean/main/syntax.lean [hunk 2]
```diff
@@ -551,4 +553,71 @@ it in the way we would expect it.
...
+
+Use the following template:
+
+```
+syntax (name := help) ...
+-- our "elaboration function" that infuses syntax with semantics
+@[command_elab help] def elabHelp : CommandElab := λ stx => Lean.logInfo "success!"
+```
+
+5. Mathlib has a ∑ operator. Create a `syntax` rule that would accept the following terms:
```
Comment (hargoniX) [inline lean/main/syntax.lean:606]: I think you should make it clear here already that you need std4 in order to use these binders (and that it is also possible to do without them)

**Review by hargoniX — CHANGES_REQUESTED** [review]
Just my remarks on the syntax chapter since that's the one I am most familiar with.

Its mostly just didactic nit picks, I think the approach is quite good in general :+1:

**Comment by lakesare** [comment]
@arthurpaulino, I think it's good to merge.

**Comment by arthurpaulino** [comment]
Amazing, thank you!

**Pull request merged**